packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,572 @@
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  return (
+    <div
+      tabIndex={0}
+      role="button"
+      onClick={onClick}
+      onKeyDown={(event) => {
+        event.key === "Enter" && onClick && onClick();
+      }}
+    >
+      <Box
+        component="a"
+        sx={{
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "center",
+          width: "32px",
+          height: "32px",
+          border: `${odysseyDesignTokens.BorderWidthMain} ${odysseyDesignTokens.BorderStyleMain} ${odysseyDesignTokens.HueNeutral300}`,
+          borderRadius: odysseyDesignTokens.BorderRadiusTight,
+          cursor: "pointer",
+          padding: odysseyDesignTokens.Spacing1,
+          "&:hover": {
+            backgroundColor: odysseyDesignTokens.HueNeutral50,
+          },
+        }}
+      >
+        <CollapseLeftIcon
+          onClick={onClick}
+          onKeyDown={onClick}
+          sx={{
+            height: "16px",
+            width: "16px",
+            alignSelf: "center",
+            color: odysseyDesignTokens.HueNeutral400,
+          }}
+        />
+      </Box>
+    </div>
+  );
+};
+
+const SideNavHeader = ({
+  navHeaderText,
+  isCollapsible,
+  onCollapse,
+}: Pick<
+  SideNavProps,
+  "navHeaderText" | "isCollapsible" | "onCollapse"
+>): ReactNode => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  return (
+    <SideNavHeaderContainer odysseyDesignTokens={odysseyDesignTokens}>
+      <Box
+        sx={{
+          marginTop: odysseyDesignTokens.Spacing2,
+        }}
+      >
+        <Heading6>{navHeaderText}</Heading6>
+      </Box>
+      {isCollapsible && <CollapseIcon onClick={onCollapse} />}
+    </SideNavHeaderContainer>
+  );
+};
+
+const SideNavListContainer = styled.ul({
+  padding: 0,
+  listStyle: "none",
+  listStyleType: "none",
+});
+
+const SectionHeader = styled("li", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  fontSize: odysseyDesignTokens.TypographySizeOverline,
+  fontWeight: odysseyDesignTokens.TypographyWeightHeadingBold,
+  color: odysseyDesignTokens.HueNeutral600,
+  paddingTop: odysseyDesignTokens.Spacing3,
+  paddingBottom: odysseyDesignTokens.Spacing3,
+  paddingLeft: odysseyDesignTokens.Spacing4,
+  textTransform: "uppercase",
+}));
+
+const SideNavItemLabelContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" &&
+    prop !== "isIconVisible" &&
+    prop !== "isDisabled",
+})<{
+  odysseyDesignTokens: DesignTokens;
+  isIconVisible: boolean;
+  isDisabled?: boolean;
+}>(({ odysseyDesignTokens, isIconVisible, isDisabled }) => ({
+  width: "100%",
+  display: "flex",
+  flexWrap: "wrap",
+  alignItems: "center",
+  fontSize: odysseyDesignTokens.TypographyScale0,
+  fontWeight: odysseyDesignTokens.TypographyWeightHeading,
+  marginLeft: isIconVisible ? odysseyDesignTokens.Spacing3 : 0,
+  "& > a": {
```

Comment:
Do we need to specify that the `a` is an immediate descendent?

Reply:
I guess I can remove this to be no too specific. I'll do that instead.